openmdao/visualization/meta_model_viewer/meta_model_visualization.py

```diff
@@ -0,0 +1,551 @@
+        for title, values in self.input_data.items():
+            slider_data = np.linspace(min(values), max(values), self.resolution)
+            self.input_data_dict[title] = slider_data
+            # Calculates the distance between slider ticks
+            slider_step = slider_data[1] - slider_data[0] 
+            self.slider_dict[title] = Slider(start=min(values), end=max(values), value=min(values), 
+            step=slider_step, title=str(title))
+
+        # Match the slider dictionary key value pairs with an on
+        # change event handler to call an update function later
+        for name, slider_object in self.slider_dict.items():
+            if name == self.x_input.value:
+                self.x_input_slider = slider_object
+                self.x_input_slider.on_change('value', self.scatter_plots_update)
+            elif name == self.y_input.value:
+                self.y_input_slider = slider_object
+                self.y_input_slider.on_change('value', self.scatter_plots_update)
+            else:
+                setattr(self, name, slider_object)
+                obj = getattr(self, name)
+                obj.on_change('value', self.update)
+
+        # Length of inputs and outputs
+        self.nx = len(self.input_list)
+        self.ny = len(self.output_list)
+
+        # Positional indicies
+        self.x_index = 0
+        self.y_index = 1
+        self.output_variable = self.output_list.index(self.output_select.value)
+
+        # Most data sources are filled with initial values
+        self.slider_source = ColumnDataSource(data=self.input_data_dict)
+        self.bot_plot_source = ColumnDataSource(data=dict(bot_slice_x=np.repeat(0, self.resolution), bot_slice_y=np.repeat(0, self.resolution)))
+        self.right_plot_source = ColumnDataSource(data=dict(left_slice_x=np.repeat(0, self.resolution), left_slice_y=np.repeat(0, self.resolution)))
+        self.source = ColumnDataSource(data=dict(x=np.repeat(0, self.resolution), y=np.repeat(0, self.resolution)))
+
+        # Text input to change the distance of reach when searching for nearest data points
+        self.scatter_distance = TextInput(value="0.1", title="Scatter Distance")
+        self.scatter_distance.on_change('value', self.scatter_input)
+        self.dist_range = float(self.scatter_distance.value)
+        
+        # Grouping all of the sliders and dropdowns into one column
+        sliders = [i for i in self.slider_dict.values()]
+        self.sliders_and_selects = row(
+            column(*sliders, self.x_input,
+            self.y_input, self.output_select, self.scatter_distance)
+        )
+
+        # Layout creation
+        self.layout = row(self.contour_data(), self.right_plot(), self.sliders_and_selects)
+        self.layout2 = row(self.bot_plot())
+        curdoc().add_root(self.layout)
+        curdoc().add_root(self.layout2)
+        curdoc().title = 'MultiView'
+
+    def make_predictions(self, data):
```

Comment:
is this supposed to be a method that a user would call? or an internal function?